lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+                             HtmlUtils.HTML('<div class="'),
+                             this.$courseImageLink.attr('class'),
+                             HtmlUtils.HTML('" tabindex="-1">'),
+                             HtmlUtils.HTML(this.$courseImageLink.html()),
+                             HtmlUtils.HTML('</div>')
+                         ).text
+                     );
+                     this.$courseTitleLink.replaceWith( // xss-lint: disable=javascript-jquery-insertion
+                         HtmlUtils.joinHtml(
+                             HtmlUtils.HTML('<span>'),
+                             this.$courseTitleLink.text(),
+                             HtmlUtils.HTML('</span>')
+                         ).text
+                     );
+                 },
+
+                 enrollError: function() {
+                     var errorMsgEl = gettext('There was an error, please reload the page and try again.');
+                     this.$dateDisplayField
+                         .find('.fa.fa-spin')
+                         .removeClass('fa-spin fa-spinner')
+                         .addClass('fa-close');
+                     this.$dateDisplayField.append(errorMsgEl);
+                     this.hideDialog($('.enroll-btn-initial'));
+                 },
+
+                 updateEnrollBtn: function() {
+                     /*
+                     This function is invoked on load, on opening the view and on changing the option on the session
+                     selection dropdown. It plays three roles:
+                     1) Enables and disables enroll button
+                     2) Changes text to describe the action taken
+                     3) Formats the confirmation popover to allow for two step authentication
+                     */
+                     var enrollText,
+                         currentSessionId = this.entitlementModel.get('currentSessionId'),
+                         newSessionId = this.$('.session-select').find('option:selected').data('session_id'),
+                         enrollBtnInitial = this.$('.enroll-btn-initial');
+
+                     // Disable the button if the user is already enrolled in that session.
+                     if (currentSessionId === newSessionId) {
+                         enrollBtnInitial.addClass('disabled');
+                         this.removeDialog(enrollBtnInitial);
+                         return;
+                     }
+                     enrollBtnInitial.removeClass('disabled');
+
+                     // Update button text specifying if the user is initially enrolling, changing or leaving a session.
+                     if (newSessionId) {
+                         enrollText = currentSessionId ? gettext('Change Session') : gettext('Select Session');
+                     } else {
+                         enrollText = gettext('Leave Current Session');
+                     }
+                     this.$('.enroll-btn-initial').text(enrollText);
```

Comment:
Will the enrollText be escaped before it's added to the DOM?

Reply:
Same as above.

Reply:
fixed